cli/view_refresh.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cli
+
+import (
+	"github.com/sourcenetwork/immutable"
+	"github.com/spf13/cobra"
+
+	"github.com/sourcenetwork/defradb/client"
+)
+
+func MakeViewRefreshCommand() *cobra.Command {
+	var name string
+	var schemaRoot string
+	var versionID string
+	var getInactive bool
+	var cmd = &cobra.Command{
+		Use:   "refresh",
+		Short: "Refresh views.",
+		Long: `Refresh views, executing the underlying query and LensVm transforms and
+persisting the results.
+
+View is refreshed as the current user, meaning results returned for all subsequent query requests
+to the view will recieve items accessible to the user refreshing the view's permissions.
```

Comment:
praise: Thanks for documenting this. 

thought: It makes me think though that this is most likely not the way we want to handle access long term. Access should be reflecting the rights of the user doing the request, not that of the user who refreshed the view. Curious to have your opinion on this.

Reply:
I like the current behaviour long term, and see very few alternatives (we could have the node-user refresh the view).

Worth remembering is that medium-long term view items will be shareable across P2P without needing to share the underlying docs.  And we have Lens stuff at play too (not all view items are sourced from any document, Lens can create their own).

The view has its own set of permissions that should be respected, although this was not handled when ACP was introduced to Defra https://github.com/sourcenetwork/defradb/issues/2018.

IMO a view is a copy of the transformed data, a user has created a view and is sharing it with who ever they see fit, as a convenient alternative to printing the results out and mailing it to people.

Reply:
> Worth remembering is that medium-long term view items will be shareable across P2P without needing to share the underlying docs.

This is specially why we can't have this behaviour long term. We can't allow some random user to access the contents of a view because the user that created the view had the rights to the documents.

Reply:
The user that created the view would need to give that random user access to it.

There is nothing we (or anyone else, really) can do to stop users sharing data they already have read access too - that is not a technical problem (see print and mail example).

Reply:
Requiring view-viewers to also have read access to the underlying data would totally undermine what I see as the primary use case - exposing data aggregates without exposing the data.

Reply:
> The user that created the view would need to give that random user access to it.

Right. Hence why I'm saying that the description is not what we want long term.

> There is nothing we (or anyone else, really) can do to stop users sharing data they already have read access to - that is not a technical problem (see print and mail example).

This tells me we have a misunderstanding. We can clarify things over a call if you'd like.